cpp/src/c_api/hits.cpp

```diff
@@ -113,40 +114,46 @@ struct hits_functor : public cugraph::c_api::abstract_functor {
       weight_t hub_score_differences{0};
       size_t number_of_iterations{0};
 
-#if 0
-      // FIXME:  Implementation will look something like this.
-      
-      if (initial_hubs_guess_ != nullptr) {
-        //
-        // Need to renumber initial_hubs_guess_vertices
-        // Need to shuffle and populate hubs
-        //
-        //  This is the original pagerank code, it will be sort of like this
-        renumber_ext_vertices<vertex_t, multi_gpu>(handle_,
-                                                   personalization_vertices_->as_type<vertex_t>(),
-                                                   personalization_vertices_->size_,
-                                                   number_map->data(),
-                                                   graph_view.get_local_vertex_first(),
-                                                   graph_view.get_local_vertex_last(),
-                                                   do_expensive_check_);
+      if (initial_hubs_guess_vertices_ != nullptr) {
+        rmm::device_uvector<vertex_t> guess_vertices(initial_hubs_guess_vertices_->size_,
+                                                     handle_.get_stream());
+        rmm::device_uvector<weight_t> guess_values(initial_hubs_guess_values_->size_,
+                                                   handle_.get_stream());
+
+        raft::copy(guess_vertices.data(),
+                   initial_hubs_guess_vertices_->as_type<vertex_t>(),
+                   guess_vertices.size(),
+                   handle_.get_stream());
+        raft::copy(guess_values.data(),
+                   initial_hubs_guess_values_->as_type<weight_t>(),
+                   guess_values.size(),
+                   handle_.get_stream());
+
+        cugraph::renumber_ext_vertices<vertex_t, multi_gpu>(handle_,
+                                                            guess_vertices.data(),
+                                                            guess_vertices.size(),
+                                                            number_map->data(),
+                                                            graph_view.get_local_vertex_first(),
+                                                            graph_view.get_local_vertex_last(),
+                                                            do_expensive_check_);
+
+        cugraph::detail::collect_vertex_values_to_local<vertex_t, weight_t, multi_gpu>(
+          handle_, guess_vertices, guess_values, hubs, graph_view.get_local_vertex_first());
```

Comment:
Is this PR pulling updates from PR #2150??? If that's the case, this PR should better be reviewed after PR #2150 gets merged (to avoid reviewing the same code twice).

Reply:
Yes.  I mentioned this in the slack thread.